### ISSUE TITLE
Sheets iterating by sheet name in alphabetical order

### ISSUE THREAD

**Issue by rps-repo** [box/spout#207]
In the past when reading sheets they iterated in the order as shown in the excel file.  Now with the latest code I am finding that no matter what order the sheets are in the file the order in which they iterate is always alphabetical.  Is this a bug or by design?

**Comment by madflow** [comment]
Hi there, the ordering is done by the Sheet-Index (a zero based numeric value). I has been changed in recently in this PR: https://github.com/box/spout/pull/165

Maybe you can post the xlsx file and the behaviour you expect?

However - I can **not** reproduce the alphabetical ordering:
```

/** @var  $writer  \Box\Spout\Writer\XLSX\Writer */
$writer = WriterFactory::create(Type::XLSX);
$writer->openToFile('wb.xlsx');
$writer->getCurrentSheet()->setName('Z-Nation');
$writer->addNewSheetAndMakeItCurrent();
$writer->getCurrentSheet()->setName('A-Team');

$writer->close();

/** @var $reader  \Box\Spout\Reader\XLSX\Reader */
$reader = ReaderFactory::create(Type::XLSX);
$reader->open('wb.xlsx');

foreach($reader->getSheetIterator() as $sheet) {
    var_dump($sheet->getName());
}
```

yields:

```
frosch ➜ spout git:(master) ✗ php test.php
string(8) "Z-Nation"
string(6) "A-Team"
```
```

/** @var  $writer  \Box\Spout\Writer\XLSX\Writer */
$writer = WriterFactory::create(Type::XLSX);
$writer->openToFile('wb.xlsx');
$writer->getCurrentSheet()->setName('A-Team');
$writer->addNewSheetAndMakeItCurrent();
$writer->getCurrentSheet()->setName('Z-Nation');

$writer->close();

/** @var $reader  \Box\Spout\Reader\XLSX\Reader */
$reader = ReaderFactory::create(Type::XLSX);
$reader->open('wb.xlsx');

foreach($reader->getSheetIterator() as $sheet) {
    var_dump($sheet->getName());
}

```

results in:

```
frosch ➜ spout git:(master) ✗ php test.php
string(6) "A-Team"
string(8) "Z-Nation"
```

**Comment by tegansnyder** [comment]
I can reproduce this issue.

Prepare a file offline in Excel with multiple sheets in it and set the path to `$existing_file_path` variable to point to it.
```
$existing_file_path = '/path/to/existing_file.xls';
$new_file_path = '/path/to/new_file.xls';

$reader = ReaderFactory::create(Type::XLSX);
$reader->open($existing_file_path);

$writer = WriterFactory::create(Type::XLSX);
$writer->openToFile($new_file_path);

// read existing file sheet by sheet
foreach ($reader->getSheetIterator() as $sheetIndex => $sheet) {

    // grab sheet name from existing file
    $existing_file_sheet_name = $sheet->getName();

    // Skip "Sheet 1"
    if ($sheetIndex !== 1) {
        $writer->addNewSheetAndMakeItCurrent();
    }

    // loop through row of existing file by the current sheet
    foreach ($sheet->getRowIterator() as $row) {

        $writer_sheet = $writer->getCurrentSheet();
        $writer_sheet->setName($existing_file_sheet_name);

        $writer->addRow($row);
    }
}

$reader->close();
$writer->close();
```

The columns will be out of order. I've created the file on Mac using Excel 2016 for OSX and running PHP 7.0.2.

I can fix the issue by commenting out the `usort` in: `Spout/Reader/XLSX/Helper/SheetHelper.php` around line 82.

```php   
        // make sure the sheets are sorted by index
        // (as the sheets are not necessarily in this order in the XML file)
        // usort($sheets, function ($sheet1, $sheet2) {
        //     return ($sheet1->getIndex() - $sheet2->getIndex());
        // });
```

**Comment by adrilo** [comment]
Thanks @tegansnyder! I was able to reproduce the issue.
@rps-repo that is definitely a bug, and the cause was pointed out by @tegansnyder.

Sheets are defined in two files:
- `[Content_Types].xml` which lists all the sheets available and where to find them. The order in this file should not matter.
- `workbook.xml` in which the order of the sheets is the correct order.

Workbook.xml looks like this:
```xml
<sheets>
	<sheet name="Sheet2" r:id="rId1" sheetId="2"/>
	<sheet name="Sheet1" r:id="rId2" sheetId="1"/>
</sheets>
```

This indicates that "Sheet1" was created first but was then moved to the 2nd position.
Spout currently exposes the sheet index, as defined in workbook.xml. I believe it needs to change as well to reflect the position in the book. So in this case, "Sheet2" would have index 0 (zero-based) and "Sheet1" index 1.
